python/chronos/src/bigdl/chronos/utils.py

```diff
@@ -30,3 +32,58 @@ def wrapped(*args, **kwargs):
             return function(*args, **kwargs)
         return wrapped
     return deprecated_decorator
+
+
+# code adaptted from https://github.com/intel/neural-compressor/
+#                    blob/master/neural_compressor/utils/utility.py#L62
+class LazyImport:
+    """
+    Example:
+        >>> from bigdl.chronos.utils import LazyImport
+        >>> forecaster = LazyImport('bigdl.chronos.forecaster.lstm_forecaster.LSTMForecaster')
+        >>> forecaster.from_tsdataset(tsdataset, ...)
+        >>> forecaster(...)
+    """
+    def __init__(self, module_name: str, pkg=None):
+        """
+        :param module_name: Import module name.
+        :param pkg: prefix path.
+        """
+        self.module_name = module_name
+        self.pkg = pkg
+
+    def __getattr__(self, name):
+        absolute_name = importlib.util.resolve_name(self.module_name, self.pkg)
+        # not reload modules
+        try:
+            return getattr(sys.modules[absolute_name], name)
+        except (KeyError, AttributeError):
+            pass
+
+        if "." in absolute_name:
+            # Split module name to prevent class name from being introduced as package
+            parent_name, _, child_name = absolute_name.rpartition('.')
+        else:
+            parent_name, child_name = absolute_name, None
+
+        try:
+            module = importlib.import_module(parent_name)
+            module = getattr(module, child_name) if child_name else module
+        except AttributeError:
+            # concatenate possible class names
+            full_module_name = parent_name+'.'+child_name if child_name else parent_name
+            spec = importlib.util.find_spec(full_module_name)
+            module = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(module)
```

Comment:
on what circumstances this branch (line 72 - 77) is needed?

Reply:
had an offline discussion, this is used for some staticmethod such as `from_tsdataset`. Please add some comment for line 70-71 and line 72-77